### DISCUSSION TITLE
How to use environment scoped vars and secrets without generating deployments?

### DISCUSSION THREAD

**Discussion by piotrekkr** [community/community#110491] [Actions]
### Select Topic Area

Question

### Body

Right now I think those are tightly coupled to deployments. Each job running with `environment: name` generate new deployment in repo. This makes using environment scoped vars and secrets pretty irritating process that pollutes repository with useless deployments that could never actually happen. 

Here is my use case:

- I have three environments like `staging`, `sandbox`, `production`.
- I have deploy workflows for each environment that triggers multiple jobs that use environment scoped vars/secrets (e.g `initialize`, `build`, `deploy`, `finalize` jobs).
- I have some workflows that do not deploy anything but also need to use env scoped vars/secrets (e.g. to connect to cloudsql instance and run some maintenance tasks on selected environment).
- I create and update deployments manually when needed because I need to set custom payload on creation and do this when some conditions are met.

As I run deploy workflow I get three deployments. First is manual one created by me, next from `build` job and one from `deploy` job. Since my manual deployment was older than the `deploy` job created it is not shown as latest deployment on env. I needed to create custom action to remove those unwanted deployments.

Same thing with other workflows that do not deploy anything but wants to access environment. Each run generates new deployment and I need to cleanup. 

Why pollute such useful feature that is environment scoped vars and secrets with deployments? Is there a way to turn off automatic deployment creation?

Thanks

**Comment by ericbf** [comment]
Duplicate of [#36919](https://github.com/orgs/community/discussions/36919)?

**Reply by piotrekkr** [reply]
Yeah. Seems like I could not find it using github search. I'll close this one
